instance/tests/models/test_instance.py

```diff
@@ -705,6 +705,7 @@ def test_provision_failed_exception_email(self):
         self.assertEqual(mime_type, "text/html")
 
 
+# pylint: disable=too-many-public-methods
```

Comment:
I don't think having many test methods on a single test class is "technical debt" just because pylint warns about it.  On the contrary, I think it's in itself a good thing.

However, I do think we have an underlying problem here, which is that the instance model does basically everything.  The fact that the functionality is split into several mix-ins helps a bit, but it doesn't fundamentally change the fact that we have one big, monolithic, strongly coupled class in the end.  That big class needs many tests, and it's good that we have many tests.  Splitting up the tests in two classes doesn't do anything to reduce technical debt.  The only thing it achieves is getting rid of a pylint directive.

I hope that this will get improved a bit by the work @bradenmacdonald is doing this week, which will split the responsibilities up into two classes.